tools/cldr-code/src/main/java/org/unicode/cldr/util/XMLFileReader.java

```diff
@@ -173,6 +173,8 @@ public static void read(
             AllHandler allHandler) {
         try {
             XMLReader xmlReader = createXMLReader(handlers, validating, allHandler);
+            // wrap the reader to insert a character stream
+            DTDInsertingReaderFactory.wrap(is);
```

Comment:
It seems like we should control this with an input parameter, rather than doing it all the time. We could change the `boolean validating` parameter from a boolean to something like:

`enum Validation {nonValidating, validating, xmlms}`

Reply:
Why does it need to be controlled?
- The purpose of the wrap function is to present a validating, `<DOCTYPE`'ed xml document to code that's expecting such
- it's only applied in cases where there's NO DOCTYPE and xmlns: is present.
- if DOCTYPE is not found, the buffer is pushed back into the stream, and reading proceeds.

Reply:
You are saying that there is no significant performance or maintenance impact of using DTDInsertingReaderFactory.wrap(is);

Reply:
BTW, seems odd that you are tossing the return value from .wrap

Reply:
> BTW, seems odd that you are tossing the return value from .wrap

It's a fluent operation, it modifies the return value.

Reply:
> You are saying that there is no significant performance or maintenance impact of using DTDInsertingReaderFactory.wrap(is);

There may be a performance impact, that's why I noted that this PR could stay open until those are known.

However, in most cases, the caller will not know whether the DTD is needed or not until the file is read.

Reply:
good point, the caller might not know. We would know, however, based on the directory.

Reply:
seems to be a statistically insignificant performance impact.